GeneratorInterface/Core/plugins/ExternalGeneratorFilter.cc

```diff
@@ -341,16 +340,25 @@ void ExternalGeneratorFilter::streamEndLuminosityBlockSummary(edm::StreamID iID,
                                                               GenLumiInfoProduct* iProduct) const {
   iProduct->mergeProduct(*streamCache(iID)->endLumiProduce(iLuminosityBlock.run()));
 
-  if (lastLumiIndex_ == iLuminosityBlock.index()) {
+  if (!luminosityBlockCache(iLuminosityBlock.index())->selectedStreamTransitionsCompleted_) {
```

Comment:
This is a race condition. Two streams could be calling `streamEndLuminosityBlockSummary` at the same time, one could be trying to read this value at the same time as one is on line 346 changing the value.

Reply:
According to our documentation (https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkGlobalModuleInterface#edm_LuminosityBlockSummaryCacheT)
> The framework guarantees that only one `streamEndLuminosityBlockSummary` is called at a time

Reply:
I agree with Matti. There is a mutex lock around those function calls:

https://cmssdt.cern.ch/dxr/CMSSW/source/FWCore/Framework/interface/global/implementors.h#267

Since it is in a lumi cache, different lumis will use different instances and different memory locations. No conflict there. And also globalEndLuminosityBlock will execute after these functions are called. This variable  isn't accessed or modified anywhere else so it should be OK.